terraform/aws/templates/base.tf

```diff
@@ -430,7 +430,20 @@ resource "aws_route_table_association" "route_internal_subnets" {
   route_table_id = "${aws_route_table.internal_route_table.id}"
 }
 
+locals {
+  ig_count = "${length(var.existing_vpc_id) > 0 ? 0 : 1}"
+  ig_id     = "${length(var.existing_vpc_id) > 0 ? data.aws_internet_gateway.default.id : join(" ", aws_internet_gateway.ig.*.id)}"
```

Comment:
This can be an issue if the existing vpc doesn't have a internet gateway. We should retrieve the internet gateway if it exists otherwise we should still create one.

Reply:
I believe this is a great idea to create IG if it does not exist, but I'm not quite sure it is possible to do. When IG does not exist, datasourse is failing with error `data.aws_internet_gateway.default: data.aws_internet_gateway.default: Your query returned no results. Please change your search criteria and try again`, and I currently do not have idea how to overcome this.
Maybe just mention in documentation, that if you're going to use existent VPC - you have to create IG on your own?

Reply:
Hi @cdutra, I believe this is a great idea to create IG if it does not exist, but I'm not quite sure it is possible to do. When IG does not exist, datasourse is failing with error `data.aws_internet_gateway.default: data.aws_internet_gateway.default: Your query returned no results. Please change your search criteria and try again`, and I currently do not have idea how to overcome this.
Maybe just mention in documentation, that if you're going to use existent VPC - you have to create IG on your own?

Reply:
Hi @cdutra, I believe this is a great idea to create IG if it does not exist, but I'm not quite sure it is possible to do. When IG does not exist, datasourse is failing with error `data.aws_internet_gateway.default: data.aws_internet_gateway.default: Your query returned no results. Please change your search criteria and try again`, and I currently do not have idea how to overcome this.
Maybe just mention in documentation, that if you're going to use existent VPC - you have to create IG on your own?

Reply:
Hi @cdutra, I believe this is a great idea to create IG if it does not exist, but I'm not quite sure it is possible to do. When IG does not exist, datasourse is failing with error `data.aws_internet_gateway.default: data.aws_internet_gateway.default: Your query returned no results. Please change your search criteria and try again`, and I currently do not have idea how to overcome this.
Maybe just mention in documentation, that if you're going to use existent VPC - you have to create IG on your own?

Reply:
Hi @cdutra, I believe this is a great idea to create IG if it does not exist, but I'm not quite sure it is possible to do. When IG does not exist, datasourse is failing with error `data.aws_internet_gateway.default: data.aws_internet_gateway.default: Your query returned no results. Please change your search criteria and try again`, and I currently do not have idea how to overcome this.
Maybe just mention in documentation, that if you're going to use existent VPC - you have to create IG on your own?